### DISCUSSION TITLE
Use dark mode colors in the widgets

### DISCUSSION THREAD

**Discussion by kflpk** [iSoron/uhabits#1069] [Feature Requests]
## Pre-submission checklist
- [x] I am submitting a bug report, not a feature request.
- [x] I have searched for similar issues, but did not find any matches.

## Description
Currently it looks like every color from the color picker has two variants, one corresponding to the light mode and one to the dark mode. The problem is that the widget colors are always in the color variant corresponding to the light mode, even if the dark mode is set in the app.
You can see the screenshots below. Keep in mind that the screenshot of the widgets was taken after the dark mode was set. 

## Steps to reproduce
1. Enable the dark mode
2. Compare the color of the habit in the app and in the widgets.

## System information
 - Phone: Mi 9 Lite
 - Phone Operating System: Android 10
 - App version: 2.0.2

## Screenshots
![scaled_dark](https://user-images.githubusercontent.com/31251884/129189443-6fd39e69-6c28-458e-a6ab-3459f27193d5.png)
![scaled_light](https://user-images.githubusercontent.com/31251884/129189431-7522742c-52b2-4804-9ba2-8c6a0c3264b4.png)
![widgets](https://user-images.githubusercontent.com/31251884/129189449-f0560d50-e1d7-456b-9826-4586ac510f02.png)

**Comment by hiqua** [comment]
> The problem is that the widget colors are always in the color variant corresponding to the light mode, even if the dark mode is set in the app.

Why is this a problem?

I can't say whether that was intentional or not, but I don't think the widget background changes depending on the mode you set in the app, does it? I assume the color change is just meant to make it more readable depending on the background chosen (dark / light). If this background stays the same (widgets), I think it makes sense to keep the same foreground color as well.

**Comment by kflpk** [comment]
>Why is this a problem?

The main problem is inconsistency. If anything, the color variants on widgets should be the dark-mode ones, since the widgets have dark background anyway. But they have the light mode variant despite it.
And it's just really annoying seeing those colors on widgets, and going to the app later, only to see that all the colors are off.

**Comment by hiqua** [comment]
Ok I see, so two main points in favor of your change:

* the user identifies the colors in the widget and in the app, so having different ones makes this harder
* the colors are not adjusted depending on the background contrary to the app, so if this change made sense in the app, it should also be in the widget

So I tend to agree with you, and as a solution I think the widget should have the same colors (including background) as the app.

**Comment by iSoron** [comment]
Thank you for bringing this up, @Yoshua-chan. I'm transferring this issue to Feature Requests since this behavior is intentional, not unexpected. We use a different set of colors for the widgets because the dark theme colors don't usually work very well with colorful (and heavily saturated) wallpapers. They also don't work well with transparent widgets.

**Reply by niktss** [reply]
But the inconsistency is pretty glaring. Also, doesn't the phone automatically darken the wallpaper when Dark Mode is enabled? This would improve contrast.

Even if not, users of widgets on busy wallpapers are probably used to non-transparent widgets because of legibility. They can just keep widget background opacity at non-0%, and if the widget background changes with dark/light mode, we can have saturated/desaturated Loop colours on it to match.
